scene/resources/packed_scene.cpp

```diff
@@ -380,7 +380,26 @@ Node *SceneState::instantiate(GenEditState p_edit_state) const {
 				//if node was not part of instance, must set its name, parenthood and ownership
 				if (i > 0) {
 					if (parent) {
-						parent->_add_child_nocheck(node, snames[n.name]);
+						bool pending_add = true;
+						if (Engine::get_singleton()->is_editor_hint()) {
+							if (parent->_get_child_by_name(snames[n.name])) {
+								// There's already a node in the same parent with the same name.
+								// This means that somehow the node was added both to the scene being
+								// loaded and another one instantiated in the former, maybe because of
+								// manual editing, or a bug in scene saving, or a loophole in the workflow
+								// (with any of the bugs possibly already fixed).
+								// Bring consistency back by letting it be assigned a non-clashing name.
+								// This simple workaround at least avoids leaks and helps the user realize
+								// something awkward has happened.
+								WARN_PRINT(vformat("Node %s clashes with another one in the scene. It will be renamed. Please fix and re-save the scene.", existing->get_name()).ascii().get_data());
```

Comment:
```suggestion
								ERR_PRINT(vformat("Node %s clashes with another one in the scene. It will be renamed. Please fix and re-save the scene '%s'.", existing->get_name(), path));
```
1. I'd say this is an error that requires action to fix and not just a warning that can be ignored
2. Just including the scene path in this error might be enough to make it clear what scene is affected
```
void (*method)(const String &) = ([](const String &p_path) {
	EditorUndoRedoManager::get_singleton()->set_history_as_unsaved(EditorNode::get_singleton()->get_editor_data().get_scene_history_id_from_path(p_path));
});
callable_mp_static(method).call_deferred(path);
```
marks the scene as unsaved, tho it is kinda eh that the call has to be deferred, otherwise I would have said definitely include that line
3. you deleted the `existing` variable that is still used here
4. the `.ascii().data()` is not needed the error macros have overloads that take `String`

Reply:
I've changed the way the error is reported so there's no need for a dependency between `packed_scene.*` and editor code.

The new change also kicks off a framework for further polishing in that regard.